go/pools/resource_pool.go

```diff
@@ -475,6 +492,27 @@ func (rp *ResourcePool) IdleClosed() int64 {
 	return rp.idleClosed.Get()
 }
 
+// RefreshTimeout returns the resource refresh timeout.
+func (rp *ResourcePool) RefreshTimeout() time.Duration {
+	return rp.refreshTimeout.Get()
+}
+
+// ExtendedRefreshTimeout returns random duration within range [RefreshTimeout, 2*RefreshTimeout)
+func (rp *ResourcePool) ExtendedRefreshTimeout() time.Duration {
+    if rp.RefreshTimeout() == 0 {
+        return 0
+    } else {
+        r := rand.New(rand.NewSource(time.Now().UnixNano()))
+        refreshTimeout := rp.RefreshTimeout()
+        return refreshTimeout + time.Millisecond * time.Duration(r.Int63n(refreshTimeout.Milliseconds()))
```

Comment:
... It looks like Go's `Duration` does handle time unit very primitively 😡

Reply:
```suggestion
        return refreshTimeout + time.Duration(r.Int63n(refreshTimeout.Nanoseconds()))
```

Why don't you use just `Nanoseconds`? I think the result is almost same, if the `r.Int63n` method returns random number with uniform distribution.

Reply:
Oh I thought the data type of Nanoseconds were float I will change it to Nanodseconds()